File: Data_structures/Tree/C++/Binary Tree.cpp
//CPP code for implementing a binary tree

#include <bits/stdc++.h> 
using namespace std; 

struct Node { 
	int data; 
	struct Node* left; 
	struct Node* right; 

	Node(int val) 
	{ 
		data = val;  
		left = NULL; 
		right = NULL; 
	} 
}; 

int main() 
{ 

	//create root
	struct Node* root = new Node(1); 
	
	root->left = new Node(2); 
	root->right = new Node(3); 
	root->left->left = new Node(4); 
	root->left->right=new Node(5);
	cout<<"Root Node is: "<<root->data<<endl;
	cout<<"Left child of root node is: "<<root->left->data<<endl;
	cout<<"Right child of root node is: "<<root->right->data<<endl;
	return 0; 
} 

File: Data_structures/Tree/Python/BinaryTree.py
# class Node:
#        def __init__(self, data):
#       self.left = None
#       self.right = None
#       self.data = data
#    def PrintTree(self):
#       print(self.data)

# root = Node(5)

# print(root.PrintTree())

class PrintTrees:
    def __init__(self, data):

        self.left = None
        self.right = None
        self.data = data

    def insert(self, data):
        if self.data:
            if data < self.data:
                if self.left is None:
                    self.left = PrintTrees(data)
                else:
                    self.left.insert(data)
            elif data > self.data:
                if self.right is None:
                    self.right = PrintTrees(data)
                else:
                    self.right.insert(data)
        else:
            self.data = data

    def PrintTree(self):
        if self.left:
            self.left.PrintTree()
        print( self.data),
        if self.right:
            self.right.PrintTree()

root = PrintTrees(27)
root.insert(14)
root.insert(35)
root.insert(31)
root.insert(10)
root.insert(19)

root.PrintTree()